### DISCUSSION TITLE
Drop the Docker daemon requirement when creating images for Spring Boot applications

### DISCUSSION THREAD

**Discussion by timtebeek** [buildpacks/community#182] [Ideas]
[Through Spring Boot](https://docs.spring.io/spring-boot/docs/2.7.1/maven-plugin/reference/htmlsingle/#build-image) we are using Cloud Native Buildpacks to create OCI images for our Spring Boot applications. This way we can create properly layered images without separately maintaining Dockerfiles.

However, as our organization adopts OpenShift, we will soon no longer be able to use [`Docker-in-Docker` in our GitLab runners](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-docker-in-docker), to access the [required Docker daemon](https://docs.spring.io/spring-boot/docs/2.7.1/maven-plugin/reference/htmlsingle/#build-image.docker-daemon).

Would it be possible to drop the Docker daemon requirement when creating images for Spring Boot applications?

I know there are similar tools out there that do not require elevated privileges or Dind to create images such as Jib and Kaniko, so in theory it could be possible. I do realize however that it might be quite the undertaking, so see this as a question for a some day maybe feature, rather than short term expectation. :)

**Comment by jromero** [comment]
The docker daemon isn't a requirement brought on by the Cloud Native Buildpacks project but instead the "platform" which in this case is the [Spring Boot maven plugin](https://github.com/spring-projects/spring-boot). You may want to consider reaching out to the maintainers of the plugin to see what options are available.

Alternatively, there are other options to the Spring Boot maven plugin to leverage buildpacks (that would yield the same result). For example, if you are using OpenShift then maybe more k8s native solutions like [kpack](https://buildpacks.io/docs/tools/kpack/) or [Tekton pipeline](https://hub.tekton.dev/tekton/pipeline/buildpacks) are the way to go.

**Reply by timtebeek** [reply]
Thanks for replying! Good to hear it might be possible already; I'll check in again with @wilkinsona, as he directed me here ([via thread](https://gitter.im/spring-projects/spring-boot?at=62c69bcc904f20479a9928ce)).
The other options sound worthwhile for my specific use case as well, I'll explore those separately from the Spring Boot plugin solution.

**Reply by wilkinsona** [reply]
Spring Boot's build pack support in Maven and Gradle strives for parity with the `pack` CLI which means that you need something listening on Docker's socket.
